### PR TITLE
Inventory shopping api

### PR DESCRIPTION
**Pull request by UTkzhang**
Added support for basic inventory and shopping apis, as well as ability for user to clear shopping cart and flash to ingredients.

Also added recipe inventory checker. If user has enough ingredients for recipe, it deducts and returns "Inventory updated, enough ingredients to proceed", if the user doesn't, it adds the missing ingredients to the shopping list and returns "Not enough ingredients, added to shopping list".

### CODE REVIEW

File: PlateUp-flask/run.py [hunk 1]
```diff
@@ -631,7 +777,7 @@ def constructRecipeTags(recipe):
     db.create_all()
     scheduler.start()
     updateRecipesToDB()
-    app.run(host='0.0.0.0')
+    app.run(host='0.0.0.0', debug=True)
```
Comment (shadow-blade-X) [inline PlateUp-flask/run.py]: Please set debug to False here

**Comment by UTkzhang** [comment]
#28 #29 #30

File: PlateUp-flask/run.py [hunk 2]
```diff
@@ -475,6 +476,151 @@ def get(self):
...
+            inventory[entry.ingredient_name] = entry.quantity
+
+        for entry in shopping_res:
+            if entry.ingredient_name not in inventory:
+                new_entry = Inventory(user_id, entry.ingredient_name, entry.quantity, entry.unit)
+                db.session.add(new_entry)
+            else:
+                inventory_entry = Inventory.query.get((user_id, entry.ingredient_name))
+                if entry.unit != inventory_entry.unit:
+                    return Response("Bad unit match while flashing to inventory.")
```
Comment (shadow-blade-X) [inline PlateUp-flask/run.py]: There should be a status code to represent error here

**Review by shadow-blade-X — CHANGES_REQUESTED** [review]

File: PlateUp-flask/run.py (hunk 2 above)
Comment (UTkzhang) [reply PlateUp-flask/run.py]: fixed

File: PlateUp-flask/run.py (hunk 1 above)
Comment (UTkzhang) [reply PlateUp-flask/run.py]: fixed

**Review by shadow-blade-X — APPROVED** [review]

**Pull request merged**